File: Java/FindtheDifference.java
public class FindtheDifference
{
	public static void main(String[] args)
	{
		new FindtheDifference();
	}
	public char solution()
	{

		String s = new String();
		String t = new String();
		int i=0;
		char[] array_s=s.toCharArray();
		char[] array_t=t.toCharArray();
		for(char c:array_t)
		{
			if(c!=array_s[i++]&&i<array_s.length)
			{
				return c;
			}
		}
		return array_t[array_t.length-1];
	}
}

File: Cplus/FindtheDifference.cpp
#include <string>
#include <unordered_map>
using namespace std;

class Solution
{
public:
	char findTheDifference(string s, string t)
	{
		unordered_map<char, int> m;
		for (auto c : t)
			++m[c];
		for (auto c : s)
		{
			if (--m[c] == 0)
				m.erase(c);
		}
		return m.begin()->first;
	}
};